pysmt/formula.py

```diff
@@ -105,20 +105,20 @@ def new_fresh_symbol(self, typename, base="FV%d"):
         return v
 
     def get_symbol(self, name):
-        if name in self.symbols:
+        try:
             return self.symbols[name]
-        return None
+        except KeyError:
+            raise UndefinedSymbolError(name)
 
     def get_all_symbols(self):
         return self.symbols.values()
 
     def get_or_create_symbol(self, name, typename):
+        if name not in self.symbols:
```

Comment:
In this way we are querying the map twice if the symbol exists. This code is quite core so this may have a significant impact.

I propose something like:
```
s = self.symbols.get(name, None)
if s is None:
    return self.create_symbol(name, typename)
...
return s
```
Otherwise (following EAFP) you can use a try.